crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -314,6 +352,21 @@ impl<'w, 's, 'a> BuildChildren for EntityCommands<'w, 's, 'a> {
         self.commands().add(AddChild { child, parent });
         self
     }
+
+    fn clear_children(&mut self) -> &mut Self {
+        let parent = self.id();
+        self.commands().add(ClearChildren { parent });
+        self
+    }
+
+    fn replace_children(&mut self, children: &[Entity]) -> &mut Self {
```

Comment:
It's generally better to use an iterator type instead of forcing someone to use a `slice`. Since it's not always guaranteed that the children entities will be in an array-like container.

```suggestion
    fn replace_children(&mut self, children: impl IntoIterator<Item = Entity>) -> &mut Self {
```

Reply:
Nice advice. But this change will get the error: the trait `smallvec::Array` is not implemented for `impl IntoIterator<Item = Entity>`, does `smallvec::Array` need to imply this trait?
Also, does `push_children(&mut self, children: &[Entity])`, `insert_children(&mut self, index: usize, children: &[Entity])` and `remove_children(&mut self, children: &[Entity])` functions also need to change the `children` parameter   type to `impl IntoIterator<Item = Entity>`?

Reply:
I believe you can use the `FromIterator` implementation on `SmallVec` ([`SmallVec::from_iter`](https://docs.rs/smallvec/latest/smallvec/struct.SmallVec.html#impl-FromIterator%3C%3CA%20as%20Array%3E%3A%3AItem%3E-for-SmallVec%3CA%3E)) instead of `SmallVec::from`.

Reply:
It's probably better to handle this in a separate PR that updates all the methods to take `IntoIterator`.